internal/cli/profile.go

```diff
@@ -157,21 +158,15 @@ func (p Profile) SetUser(publicAPIKey, privateAPIKey string) {
 	p.SetString(keyPrivateAPIKey, privateAPIKey)
 }
 
-// Session is the CLI profile session
-type Session struct {
```

Comment:
This may be a bit controversial, but I removed the session from profile and put it in client.go. The intent was to allow a user to create an Auth client that had session information so I moved session to the realm client since we would have a circular import if we imported profile.Session in the realm client. Don't mind a different approach at all here

Reply:
The package cycle callout makes sense, I think this is a fine approach for now (we'll be doing something similar with `telemetry` and the profile as well).

My only suggestions around this would be:
1. Move this to `realm/auth.go` and replace `AuthResponse` with this struct (keeping its name `Session`)
1. Make sure to add the `json` tags there for unmarshaling the server response, and update some variable names in the login code below
1. Keep `session` as a field on the `client`, but just as a value (don't use a pointer)

I'll touch on the validation part of all this (and how we should handle auth/refresh) in a later comment